src/main/cpp/tests/format_float.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cast_string.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/strings/convert/convert_floats.hpp>
+
+#include <limits>
+#include <rmm/device_uvector.hpp>
```

Comment:
The headers should be grouped by a "near to far" order: local headers first, then `cudf_test`, then `cudf/`, then `thrust`, then `rmm`.
```suggestion
#include <rmm/device_uvector.hpp>

#include <limits>
```